src/main/java/core/rating/RatingPredictionModel.java

```diff
@@ -1712,11 +1712,14 @@ protected double calcLongshots(@NotNull Lineup lineup, int minute) {
         var sumSetPieces = 0.;
         var n = 0;
         for (var p : lineup.getFieldPositions()) {
+            if ( p.getRoleId() == keeper) continue;
             var player = p.getPlayer();
+
             if (player != null) {
+                var loyalty = calcLoyalty(player);
                 n++;
-                sumScoring += calcSkillRating(player.getSkill(SCORING));
-                sumSetPieces += calcSkillRating(player.getSkill(SETPIECES));
+                sumScoring += loyalty + calcSkillRating(player.getSkill(SCORING));
```

Comment:
Is that described somewhere, the impact of loyalty on long shots rating, or is it empirical?